calico_rkt/calico_rkt.py

```diff
@@ -1,198 +1,212 @@
-    return endpoint, ip
-
-def _container_add(hostname, orchestrator_id, container_id, netns_path, interface, client):
+                      orchestrator_id=ORCHESTRATOR_ID,
+                      container_id=container_id)
+
+    # Delete profile if only member
+    if _datastore_client.profile_exists(net_name) and \
+       len(_datastore_client.get_profile_members(net_name)) < 1:
+        try:
+            _log.info(
+                "Profile %s has no members, removing from datastore" % net_name)
+            _datastore_client.remove_profile(net_name)
+        except:
+            _log.error(
+                "Cannot remove profile %s: Profile cannot be found." % container_id)
+            sys.exit(1)
+
+
+def _create_calico_endpoint(container_id, netns_path, interface, subnet):
     """
-    Add a container to Calico networking with the given IP
-    Return Endpoint object
+    Configure the Calico interface for a pod.
+    Return Endpoint and IP
     """
+    _log.info('Configuring Calico networking.')
+
     try:
-        _ = client.get_endpoint(
-            hostname=hostname,
-            orchestrator_id=orchestrator_id,
-            workload_id=container_id
-        )
+        _ = _datastore_client.get_endpoint(hostname=HOSTNAME,
+                                           orchestrator_id=ORCHESTRATOR_ID,
+                                           workload_id=container_id)
     except KeyError:
         # Calico doesn't know about this container.  Continue.
         pass
     else:
-        print_stderr("This container has already been configured with Calico Networking.")
+        _log.error(
+            "This container has already been configured with Calico Networking.")
         sys.exit(1)
 
-    # Allocate and Assign ip address through IPAM Client
-    ip = _allocate_IP()
+    endpoint, ip = _container_add(hostname=HOSTNAME,
+                                  orchestrator_id=ORCHESTRATOR_ID,
+                                  container_id=container_id,
+                                  netns_path=netns_path,
+                                  interface=interface,
+                                  subnet=subnet)
+
+    _log.info('Finished configuring network interface')
+    return endpoint, ip
+
+
+def _container_add(hostname, orchestrator_id, container_id, netns_path, interface, subnet):
+    """
+    Add a container to Calico networking
+    Return Endpoint object and newly allocated IP
+    """
+    # Allocate and Assign ip address through IPAM _datastore_client
+    pool = _generate_pool(subnet)
+    ip = _allocate_ip(pool)
 
     # Create Endpoint object
     try:
-        ep = client.create_endpoint(HOSTNAME, ORCHESTRATOR_ID,
-                                      container_id, [IPAddress(ip)])
+        ep = _datastore_client.create_endpoint(HOSTNAME, ORCHESTRATOR_ID,
+                                               container_id, [ip])
     except AddrFormatError:
```

Comment:
Seems odd to me that we'd get an AddrFormatError here. Any clue why this is here?